### DISCUSSION TITLE
Error during schema generation for generic model with enum field

### DISCUSSION THREAD

**Discussion by SnakeOilSalesman** [fastapi/fastapi#10416] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import enum
from typing import Annotated
from typing import Generic
from typing import TypeVar

from fastapi import Depends
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class Response(BaseModel):
    name: str
    age: int
    address: str


FieldsSet = TypeVar('FieldsSet', bound=enum.StrEnum)


class SortingParams(BaseModel, Generic[FieldsSet]):
    sort_by: FieldsSet | None = None


FieldsOptions = enum.StrEnum('FieldsOptions', {f.capitalize(): f for f in Response.model_fields})


@app.get('/', response_model=Response)
def index(
    sort: Annotated[SortingParams[FieldsOptions], Depends()],
) -> dict[str, str | int]:
    print(sort)
    return {'name': 'John Doe', 'age': 21, 'address': 'Penny Lane'}
```

### Description

I'm trying to implement an approach, using which the following will be possible:

- Using Enum and Generic pydantic model, sorting query parameter options will be generated
- Those options will be used during parameter validation
- Generated options can be seen in OpenAPI document (Swagger or redoc)
- Using Generic model, I want to have a common interface for sorting of other endpoints of my application

The problem is that using code from example, it works in general, but Swagger and redoc are not able to generate the schema.

Since this example works:
```Python
SortingParams[FieldsOptions].model_json_schema()
```
and this one doesn't:
```Python
SortingParams.model_json_schema()
```
my assumption - it stumbles in attempt to generate the schema of generic model `SortingParams`.

As I understand, this is expected behaviour for pydantic.

Will be happy to hear any suggestions.

### Operating System

Linux

### Operating System Details

Ubuntu 22.04.3 LTS

### FastAPI Version

0.103.1

### Pydantic Version

2.4.1

### Python Version

Python 3.11.5

**Comment by hgalytoby** [comment]
I have the same problem.

FastAPI Version
0.110.0

Pydantic Version
1.10

Python Version
Python 3.10.10


It's working.
```py
from enum import Enum
from typing import TypeVar, Generic, Optional

from fastapi.params import Query
from pydantic.generics import GenericModel

SortFieldType = TypeVar("SortFieldType", bound=Enum)


class SortField(str, Enum):
    URI = "uri"


class BaseQuery(GenericModel, Generic[SortFieldType]):
    sort_field: Optional[SortFieldType] = Query(
        default=None,
    )
    sort: Optional[bool] = Query(
        default=None,
    )


class QueryModel(BaseQuery[SortField]):
    ...


query = QueryModel(sort_field=SortField.URI)
```


An error occurs when accessing `/docs` after FastAPI has been activated.

`TypeError: issubclass() arg 1 must be a class`

Can anyone help?

**Comment by YuriiMotov** [comment, marked as answer]
This is a known issue with optional enums (actually, not only enums) in Query params.
You can solve it using `SkipJsonSchema[None]` instead of `None` in type annotation

```
import enum
from typing import Annotated, Generic, TypeVar

from fastapi import Depends, FastAPI
from pydantic import BaseModel
from pydantic.json_schema import SkipJsonSchema


app = FastAPI()

class Response(BaseModel):
    name: str
    age: int
    address: str

FieldsSet = TypeVar("FieldsSet", bound=enum.StrEnum)

FieldsOptions = enum.StrEnum(
    "FieldsOptions", {f.capitalize(): f for f in Response.model_fields}
)

class SortingParams(BaseModel, Generic[FieldsSet]):
    sort_by: FieldsSet | SkipJsonSchema[None] = None

@app.get("/", response_model=Response)
def index(sort: Annotated[SortingParams[FieldsOptions], Depends()]):
    print(sort)
    return {"name": "John Doe", "age": 21, "address": "Penny Lane"}
```
